lib/routes/content.js

```diff
@@ -144,60 +153,136 @@ const routes = application => {
         return next();
       }
 
-      return database(`content-type--${type.id}`).where('id', req.params.id).then(rows => {
-        if (rows.length < 1) {
-          _.set(req.session, '404', {
-            message: config.content.messages.missing.id.replace('%type', req.params.type).replace('%id', req.params.id),
-            safe: `/${config.content.base}/${req.params.type}`,
-          });
+      // no id in url, so 404
+      if (!req.params.id) {
+        _.set(req.session, '404', {
+          message: config.content.messages.missing.id.replace('%type', req.params.type).replace('%id', req.params.id),
+          safe: `/${config.content.base}/${req.params.type}`,
+        });
 
-          return next();
-        }
+        return next();
+      }
+
+      // no revision in url, so create search to find latest revision
+      if (!req.params.revision || !Number.isInteger(req.params.revision)) {
+        revisionSearch = database(`content-type--${type.id}`)
```

Comment:
@Snugug these are promises. [See below for usage as promises](https://github.com/scottnath/punchcard/blob/e4112ffadf3ea1274c8a3161dbabf63bd1a3e949/lib/routes/content.js#L208)
I'm giving `revisionSearch` either a promise that searches for a revision if one isn't provided, or a promise to test that the revision they provided actually exists (line 175 below)